clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala

```diff
@@ -209,24 +209,17 @@ class LakeFSAllRangesInputFormat extends LakeFSBaseInputFormat {
     if (StringUtils.isBlank(storageNamespace)) {
       storageNamespace = apiClient.getStorageNamespace(repoName, StorageClientType.HadoopFS)
     }
-    var lakeFSMetadataURI = URI.create(storageNamespace)
-    lakeFSMetadataURI = new URI(
-      lakeFSMetadataURI.getScheme,
-      lakeFSMetadataURI.getUserInfo,
-      lakeFSMetadataURI.getHost,
-      lakeFSMetadataURI.getPort,
-      lakeFSMetadataURI.getPath + (if (lakeFSMetadataURI.getPath.endsWith("/")) "_lakefs"
-                                   else "/_lakefs"),
-      null,
-      null
-    )
-    val fs = FileSystem.get(lakeFSMetadataURI, conf)
-    val splits = new ListBuffer[InputSplit]()
-    val metadataPath = new Path(lakeFSMetadataURI.toString)
+    val namespaceURI = URI.create(storageNamespace)
+    val fs = FileSystem.get(namespaceURI, conf)
+    fs.getStatus(new Path(namespaceURI)) // Will throw an exception if namespace doesn't exist
+
+    val metadataURI = namespaceURI.resolve("_lakefs")
```

Comment:
I love that the code is much cleaner here!
IIRC, resolve will only get the desired result here if the storage namespace ends with a "/" here, so better double-check this. If it's the case, consider adding it if needed before creating the URI.